git_repo/repo.py

```diff
@@ -215,7 +215,7 @@ def set_repo_slug(self, repo_slug, auto=False):
 
             # This needs to be manually plucked because otherwise it'll be unset for some commands.
             service = RepositoryService.get_service(None, self.target)
-            if len(namespace) > service._max_nested_namespaces:
+            if service._max_nested_namespaces and len(namespace) > service._max_nested_namespaces:
```

Comment:
I dont think this is necessary.  Just set `max_nested_namespaces` to be 99 if it truly is unlimited in gerrit (I doubt that; I am sure gerrit will break somewhere if the project list has 99 subtrees)

Reply:
does it actually support project namespaces ?

Reply:
Yup; Wikimedia has at least 5 levels of namespaces that I have seen.  They might use more.

Reply:
Okay. `max_nested_namespaces` will be changed to 99